From 2239772b8e7d9553c71056a93965c04fe6417e66 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 15 Sep 2026 22:53:10 +0200
Subject: [PATCH] avformat/asfdec_f: index keyframes at the packet position in
 asf_read_pts()

Fixes: assertion failure
Found during triage/review of the security report M3Ib1Zm6OX8V
---
 libavformat/asfdec_f.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/libavformat/asfdec_f.c b/libavformat/asfdec_f.c
index eb30283ce16a..212ad449649e 100644
--- a/libavformat/asfdec_f.c
+++ b/libavformat/asfdec_f.c
@@ -1447,7 +1447,6 @@ static int64_t asf_read_pts(AVFormatContext *s, int stream_index,
     FFFormatContext *const si = ffformatcontext(s);
     ASFContext *asf     = s->priv_data;
     AVPacket pkt1, *pkt = &pkt1;
-    ASFStream *asf_st;
     int64_t pts;
     int64_t pos = *ppos;
     int i;
@@ -1477,15 +1476,10 @@ static int64_t asf_read_pts(AVFormatContext *s, int stream_index,
         if (pkt->flags & AV_PKT_FLAG_KEY) {
             AVStream *st = s->streams[pkt->stream_index];
 
-            asf_st = &asf->streams[st->id];
-
-//            assert((asf_st->packet_pos - s->data_offset) % s->packet_size == 0);
-            pos = asf_st->packet_pos;
-            av_assert1(pkt->pos == asf_st->packet_pos);
-
+            pos = pkt->pos;
             av_add_index_entry(st, pos, pts, pkt->size,
                                pos - start_pos[st->id] + 1, AVINDEX_KEYFRAME);
-            start_pos[st->id] = asf_st->packet_pos + 1;
+            start_pos[st->id] = pos + 1;
 
             if (pkt->stream_index == stream_index) {
                 av_packet_unref(pkt);
